pymeasure/instruments/nanovna/nanovna.py

```diff
@@ -0,0 +1,165 @@
+            write_termination='\r',
+            **kwargs
+        )
+
+    def read(self):
+        """
+        Reads from the instrument including the correct termination characters
+        """
+        ret = self.adapter._read_echo()
+        return ret
+
+    def write(self, command):
+        """
+        Writes to the instrument including the device address
+
+        :param command: command string to be sent to the instrument
+        """
+        super().write(command)
+
+    def ask(self, command, query_delay=0):
+        """Write a command to the instrument and return the read response.
+
+        :param command: Command string to be sent to the instrument.
+        :param query_delay: Delay between writing and reading in seconds.
+        :returns: String returned by the device without read_termination.
+        """
+        self.write(command)
+        self.wait_for(query_delay)
+        return self.adapter._read_echo()
+
+    help = Instrument.measurement(
+        'help',
+        """Returns list of available commands. """)
+
+    info = Instrument.measurement(
+        'info',
+        """Returns device information. """)
+
+    pause = Instrument.measurement(
+        'pause',
+        """Pauses acquisition. Returns nothing. """)
+
+    resume = Instrument.measurement(
+        'resume',
+        """Resume acquisition.  Returns nothing. """)
+
+    frequencies = Instrument.measurement("frequencies",
+                                         docs=""" Returns frequencies as a list
+                                         of floats. """,
+                                         # get_process=_process_frequencies,
+                                         )
+
+    data = Instrument.measurement("data",
+                                  docs=""" Returns a list of strings containing
+                                  the complex numbers. """,
+                                  )
+
+    trace = Instrument.measurement("trace",
+                                   docs=""" Gets the trace settings. """,
+                                   # TODO - make this into a control,
+                                   )
+
+    power = Instrument.setting('power %d',
+                               """ Sets the output power. """,
+                               validator=strict_discrete_set,
+                               values=[-1, 0, 1, 2, 3],
+                               )
+
+    sweep = Instrument.control('sweep',
+                               'sweep %s',
+                               """Sets the sweep details.  Input is a string
+                               containing 3 integers separated by spaces.
+                               First int is the start frequency (in Hz).
+                               Second int is the strop frequency (in Hz).
+                               Third int is the number of points in the sweep.
+                               Example:  '13000000 16000000 101'  """,
+                               )
```

Comment:
Why don't you accept a tuple of numbers instead of the string?
```suggestion
    sweep = Instrument.control('sweep',
                               'sweep %i %i %i',
                               """Set the sweep details.  Input is a tuple
                               containing 3 integers.
                               First int is the start frequency (in Hz).
                               Second int is the strop frequency (in Hz).
                               Third int is the number of points in the sweep.
                               Example:  :code:`13000000 16000000 101`.  """,
                               )
```
That way, a user might set `inst.sweep = 130000, 1600000, 101` with integers and not with a string.

By the way, a docstring should be in imperative form, so it is "Set" and not "Sets"

Reply:
Very clever.  Thanks!